Nexpo/Repositories/TicketRepository.cs

```diff
@@ -58,8 +58,17 @@ public async Task<Ticket> GetByCode(Guid code)
 
         public async Task Add(Ticket ticket)
         {
-            _context.Tickets.Add(ticket);
-            await _context.SaveChangesAsync();
+            using (var dbContextTransaction = _context.Database.BeginTransaction())
+            {
+                var events = _context.Events.Where(e => e.Id == ticket.EventId).ToList();
+                   
+                if(events[0].TicketCount < events[0].Capacity)
+                {
+                    _context.Tickets.Add(ticket);
+                    await _context.SaveChangesAsync();
+                }
```

Comment:
Is there a reason to get a list? I have not run this to confirm but I suspect that events[0].TicketCount would be the default value since it is not set in this query. You need to create a join of the two tables and populate the count. The easiest way I can think of is to:

1. Inject the EventRepository in the TicketRepository
2. Use the Get(id) method to get a complete Event
3. Check that the capacity has not been reached

```suggestion
                var event = await _eventRepo.Get(ticket.EventId);
                
                // Don't add ticket if limit is reached
                if (event.TicketCount == event.Capacity) {
                     return false;
                }
                
                _context.Tickets.Add(ticket);
                await _context.SaveChangesAsync();
                dbContextTransaction.Commit();
                return true;
```

My suggestion is to implement something similar to the above. Notice that you will need to change the method signature to return a boolean that denotes if the add was a success. You could then check the result in the controller and return an error status code if the add failed.

(The above code is not tested and just written as a suggestion, I have not tried it since I don't have my dev env setup)